components/style/values/generics/position.rs

```diff
@@ -0,0 +1,283 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Generic types for CSS handling of specified and computed values of
+//! [`position`](https://drafts.csswg.org/css-backgrounds-3/#position)
+
+use cssparser::Parser;
+use parser::{Parse, ParserContext};
+use std::fmt;
+use style_traits::ToCss;
+use values::HasViewportPercentage;
+use values::computed::{ComputedValueAsSpecified, Context, ToComputedValue};
+use values::specified::{LengthOrPercentage, Percentage};
+
+define_css_keyword_enum!{ Keyword:
+    "center" => Center,
+    "left" => Left,
+    "right" => Right,
+    "top" => Top,
+    "bottom" => Bottom,
+    "x-start" => XStart,
+    "x-end" => XEnd,
+    "y-start" => YStart,
+    "y-end" => YEnd
+}
+
+add_impls_for_keyword_enum!(Keyword);
+
+impl Keyword {
+    #[inline]
+    /// Check whether this keyword indicates the opposite sides. This doesn't actually check the side,
+    /// and we don't have to, since that's done during parsing.
```

Comment:
I'm not sure what this even means, what does `is_other_side` mean? The opposite side with respect of what?